integration/spark2/src/main/scala/org/apache/carbondata/indexserver/IndexServer.scala

```diff
@@ -0,0 +1,136 @@
+
+  private val serverPort: Int = CarbonProperties.getInstance().getIndexServerPort
+
+  /**
+   * Getting sparkSession from ActiveSession because in case of embedded mode the session would
+   * have already been created whereas in case of distributed mode the session would be
+   * created by the main method after some validations.
+   */
+  private var sparkSession: SparkSession = SparkSQLUtil.getSparkSession
+
+  def getSplits(request: DistributableDataMapFormat): Array[ExtendedBlocklet] = {
+    val splits = new DistributedPruneRDD(sparkSession, request).collect()
+    DistributedRDDUtils.updateExecutorCacheSize(splits.map(_._1).toSet)
+    splits.map(_._2)
+  }
+
+  override def invalidateCache(request: DistributableDataMapFormat): Unit = {
+    val splits = new DistributedPruneRDD(sparkSession, request).collect()
+    DistributedRDDUtils.updateExecutorCacheSize(splits.map(_._1).toSet)
+    if (request.isJobToClearDataMaps) {
+      DistributedRDDUtils.invalidateCache(request.getCarbonTable.getTableUniqueName)
+    }
+  }
+
+  override def showCache(tableName: String = ""): Seq[String] = {
+    new DistributedShowCacheRDD(sparkSession, tableName).collect().toSeq
+  }
+
+  def main(args: Array[String]): Unit = {
+    if (serverIp.isEmpty) {
+      throw new RuntimeException(s"Please set the server IP to use Index Cache Server")
+    } else if (!isDistributedPruning) {
+      throw new RuntimeException(
+        s"Please set ${ CarbonCommonConstants.CARBON_ENABLE_INDEX_SERVER }" +
+        s" as true to use index server")
+    } else {
+      val (storePath, masterURL) = if (args.length == 1) {
+        (args.head, "local")
+      } else if (args.length == 2) {
+        (args.head, args.tail.head)
+      } else {
+        (null, "local")
+      }
+      sparkSession = createCarbonSession(storePath, masterURL)
+      LOGGER.info("Starting Index Cache Server")
+      val conf = new Configuration()
+      val server: RPC.Server = new RPC.Builder(conf).setInstance(this)
+        .setBindAddress(serverIp)
+        .setPort(serverPort)
+        .setProtocol(classOf[ServerInterface]).build
+      server.start()
+      sparkSession.sparkContext.addSparkListener(new SparkListener {
+        override def onApplicationEnd(applicationEnd: SparkListenerApplicationEnd): Unit = {
+          LOGGER.info("Spark Application has ended. Stopping the Index Server")
+          server.stop()
+        }
+      })
+      LOGGER.info(s"Index cache server running on ${ server.getPort } port")
+    }
+  }
+
+  private def createCarbonSession(storePath: String, masterURL: String): SparkSession = {
+    import org.apache.spark.sql.CarbonSession._
+    val master = System.getProperty("spark.master.url", masterURL)
+    if (master == null) {
+      throw new InvalidConfigurationException("Spark master URL is not set.")
+    }
+    val spark = SparkSession
+      .builder().config(new SparkConf())
```

Comment:
Is there a possibility of passing other configurations like number of executors, cores, etc ?..If yes then how those values will be passed?

Reply:
yes they can be set in spark-defaults.conf.